ext/bindings/message.cpp

```diff
@@ -18,6 +18,10 @@ Message::Message(const std::string& data) {
   _msg = mb.build();
 }
 
+Message Message::fromMessage(const pulsar::Message& msg) {
```

Comment:
I think the return type here is the source of the error you mentioned for `from_message` ruby-side. should return a `Message::ptr`; pair with `return Message::ptr(new Message(message))` on the following line. this'll allocate the object being returned off the stack and wrap it up into Rice's ruby-aware memory management

Reply:
ah ok, that makes sense. I will try and finesse it into working

Reply:
that still gives the same error, sadly

Reply:
moved this method into ruby-land so the error is moot 👍🏻